### PR TITLE
Update 2.3 Release Notes

### PR DESCRIPTION
**Pull request by jakirkham**
Adds some content that was left out of the 2.3 release notes that has been included since the 2.2.0 release.

TODO:
* [ ] Add unit tests and/or doctests in docstrings
* [ ] Add docstrings and API docs for any new/modified user-facing classes and functions
* [ ] New/modified features documented in docs/tutorial.rst
* [ ] Changes documented in docs/release.rst
* [ ] Docs build locally (e.g., run ``tox -e docs``)
* [ ] AppVeyor and Travis CI passes
* [ ] Test coverage is 100% (Coveralls passes)

### CODE REVIEW

**Comment by jakirkham** [comment]
I think this is ready for review and merge if someone has a chance to take a look.

cc @zarr-developers/core-devs

File: docs/release.rst [hunk 1]
```diff
@@ -36,7 +38,8 @@ Enhancements
 
 * Adds the MongoDB-backed :class:`zarr.storage.MongoDBStore` class enabling a
   MongoDB database to be used as the backing store for an array or group.
-  By :user:`Joe Hamman <jhamman>`, :issue:`299`, :issue:`372`.
+  By :user:`Joe Hamman <jhamman>`, :user:`John Kirkham <jakirkham>`,
+  :issue:`299`, :issue:`372`, :issue:`401`.
```
Comment (jhamman) [inline docs/release.rst]: can we add @nbren12 to the list of contributors here? He prototyped this and though his work doesn't show up in the commit record, he deserves 95% of the credit.

**Comment by nbren12** [comment]
Thanks Joe. I feel like the 80-20 rule applies here though. Writing the
prototype was the easy part.

File: docs/release.rst (hunk 1 above)
Comment (alimanfoo) [reply docs/release.rst]: +1 to adding @nbren12, I think we should credit all people who contributed towards it.

Comment (jakirkham) [reply docs/release.rst]: Yep, agreed. Sorry this was an oversight.

**Comment by jakirkham** [comment]
Have responded to the reviews. Please let me know if there is anything else needed 🙂

**Comment by jakirkham** [comment]
Thanks all. Will go ahead and cut the release now.

**Pull request merged**